api/base/serializers.py

```diff
@@ -80,6 +81,22 @@ def to_internal_value(self, data):
         return super(TypeField, self).to_internal_value(data)
 
 
+class TargetTypeField(ser.CharField):
+    """
+    Enforces that the related resource has the correct type
+    """
+    def __init__(self, target_type, **kwargs):
```

Comment:
Since you're already addressing this as a kwarg [below](https://github.com/CenterForOpenScience/osf.io/pull/4439/files#diff-7a6d21a50624153d15e63a5f643401c7R239) below, consider making it one here.